google/mozc
# Also deploy `mozc_tip64.dll.pdb`

This closes  #1081.

## Context
One of the biggest challenge while investigating the following issue has been that it is reliably reproducible only on certain users' environments.

 * https://github.com/google/mozc/issues/1076

One idea to accelerate the debugging is to ship debug symbol file for Mozc TIP DLL (e.g. `mozc_tip64.dll`) to users. Then we can ask the reporter to use tools like [Process Explorer](https://learn.microsoft.com/en-us/sysinternals/downloads/process-explorer) and [Process Hacker](https://processhacker.sourceforge.io) to dump the call stack of the thread that got stuck.

## Description
With this commit we actually start deploying `mozc_tip64.dll.pdb` to users.

The storage impact looks to be acceptable overall as long as we [strip private symbols](https://learn.microsoft.com/en-us/windows-hardware/drivers/debugger/public-and-private-symbols). Here are actual values taken in my local environment.

 * `Mozc64.msi`: + 584 kB
 * `mozc_tip64.dll.pdb`: + 6,116 kB

## Caveats

This commit does not fully take care of Bazel build. While the symbol file is actually deployed, there remain the following known
issues only in Bazel build.

 * private symbols are not yet stripped out.
 * the symbol file name embedded in `mozc_tip64.dll` is `mozc_tip.pdb` rather than `mozc_tip64.dll.pdb`.

## Issue IDs

* https://github.com/google/mozc/issues/1081

## Steps to test new behaviors (if any)
A clear and concise description about how to verify new behaviors (if any).
 - OS: Windows 11 23H2
 - Steps:
   1. Install Mozc
   2. Open `%ProgramFiles(x86)%\Mozc\mozc_tip64.dll` with `peview.exe` included in [Process Hacker](https://processhacker.sourceforge.io).

Here is the expected behavior.
![Image](https://github.com/user-attachments/assets/87272288-15e2-4c6e-94f6-606aed8046e3)
